lib/active_model/serializer.rb

```diff
@@ -177,6 +177,10 @@ def json_key
       end
     end
 
+    def as_json
+      self.class.adapter.new(self).as_json
```

Comment:
This doesn't make sense to me, feature wise.  You should never be calling to_json/as_json on a serializer.  The serializer's job is to serialize the resource. The adapter formats the serialized resource per json/json-api whatever, and is the actual object that gets to_json/as_json called on it.

I'm :-1: on this unless I'm totally misunderstanding something.  See http://www.benjaminfleischer.com/2015/06/02/understanding-rails-model-serializers/